eng/liveBuilds.targets

```diff
@@ -14,7 +14,8 @@
 
   <!-- Set up default live asset paths if no overrides provided. -->
   <PropertyGroup>
-    <CoreCLRArtifactsPath Condition="'$(CoreCLRArtifactsPath)' == ''">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'coreclr', '$(TargetOS).$(TargetArchitecture).$(CoreCLRConfiguration)'))</CoreCLRArtifactsPath>
+    <CoreCLRArtifactsPath Condition="'$(CoreCLRArtifactsPath)' == '' and '$(TargetsAppleMobile)' != 'true'">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'coreclr', '$(TargetOS).$(TargetArchitecture).$(CoreCLRConfiguration)'))</CoreCLRArtifactsPath>
+    <CoreCLRArtifactsPath Condition="'$(CoreCLRArtifactsPath)' == '' and '$(TargetsAppleMobile)' == 'true'">$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts', 'bin', 'coreclr', '$(HostOS).$(BuildArchitecture).$(CoreCLRConfiguration)'))</CoreCLRArtifactsPath>
```

Comment:
Thinking about this more, I think we already have a pattern for the problem this is trying to solve.

When we build for arm64 platforms (be it Windows/Linux/Mac), we currently have a problem that the build machines are always x64 and the ILC we just built is not usable. We build one more ILC that matches the architecture of the build machine and use that one to build the tests.

This other ILC is built with the [ILCompiler_crossarch](https://github.com/dotnet/runtime/blob/36439c510b779103a4a8066359d0d63dc003eed3/src/coreclr/tools/aot/ILCompiler/ILCompiler_crossarch.csproj) project.

We have a pointer to it in liveBuilds:

https://github.com/dotnet/runtime/blob/36439c510b779103a4a8066359d0d63dc003eed3/eng/liveBuilds.targets#L27-L28

iDevices are in a similar situation except we don't just change the architecture but also the OS. I think we could make it fit this scenario instead of having to build things twice.

The relevant PRs that added this were #64373 and #82307.

Reply:
The problem on iDevices is similar in that we don't have a usable ILC but we can just build it because the infra already exists.

Reply:
I've tried to use the cross build for Native AOT runtime tests on iOS in [Test cross build · dotnet/runtime@7d46128 (github.com)](https://github.com/dotnet/runtime/commit/7d461286cd7ac3dab79228b50c87a71ae3455b7a). When I test it locally with the following command, it fails when generating native code with ld: `unknown option: -EL`.

```
./build.sh -os ios -arch arm64 -cross -s clr.alljits+clr.tools+clr.nativeaotlibs+clr.nativeaotruntime+libs
```

Reply:
/cc @filipnavara @ivanpovazan

Reply:
I am not sure which build paths are taken when passing `-cross` with `clr.alljits+clr.tools` subsets, but I would try building only those two subsets to be sure if they are throwing.

Reply:
The libjitinterface_arm64.dylib is created in the default directory, which is `artifacts/bin/coreclr/ios.arm64.Release/libjitinterface_arm64.dylib`. However, when cross-compilation is used, it is expected to be in the different directory `artifacts/bin/coreclr/ios.arm64.Release/osx/arm64/libjitinterface_arm64.dylib`.

When copied manually to the target directory, the build works.

Reply:
The one in `artifacts/bin/coreclr/ios.arm64.Release/libjitinterface_arm64.dylib` is likely built for the target (iOS in this case). The expected one in `artifacts/bin/coreclr/ios.arm64.Release/osx/arm64/libjitinterface_arm64.dylib` should be built for the host. If it's not getting generated, something is wrong with the conditions for native build somewhere. I'm not 100% fluent in all of those.

Can you try building this under Rosetta command prompt to basically do a crossbuild from x64 macOS to arm64 iOS? I think a bunch of the crossbuild conditions assume we're going to from x64 host to arm64 target and will likely be written that way. Getting _something_ working might help narrow down what exactly is wrong. I don't have better pointers unfortunately.

Reply:
> I've created a tracking issue #87764.

Would be nice to address it here - it looks like the necessity of doing two builds and then collecting things from two different locations accounts for many of the diffs in this pr - it's usually harder to fully eradicate those later (I've seen many vestigial remains of old approaches that were missed when deleting them at a later point in the past and it feels like this will be one of those).

Reply:
That sounds reasonable -- I suggest we address it in a separate PR (to allow easier testing and code review) before we merge this PR and the functional tests in https://github.com/dotnet/runtime/pull/87773.